modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductSpecificationResourceImpl.java

```diff
@@ -62,6 +81,34 @@ public Page<ProductSpecification> getProductIdProductSpecificationsPage(
 			id, contextAcceptLanguage.getPreferredLocale(), pagination);
 	}
 
+	@Override
+	public ProductSpecification getProductSpecification(Long id)
+		throws Exception {
+
+		CPDefinitionSpecificationOptionValue
+			cpDefinitionSpecificationOptionValue =
+				_cpDefinitionSpecificationOptionValueService.
+					getCPDefinitionSpecificationOptionValue(id);
+
+		return _toProductSpecification(
+			cpDefinitionSpecificationOptionValue.
+				getCPDefinitionSpecificationOptionValueId());
+	}
+
+	@Override
+	public ProductSpecification patchProductSpecification(
+			Long id, ProductSpecification productSpecification)
+		throws Exception {
+
+		CPDefinitionSpecificationOptionValue
+			cpDefinitionSpecificationOptionValue = _updateProductSpecification(
```

Comment:
this method is running this code 
`return cpDefinitionSpecificationOptionValueService.
			updateCPDefinitionSpecificationOptionValue(
				cpDefinitionSpecificationOptionValue.
					getCPDefinitionSpecificationOptionValueId(),
				_getCPOptionCategoryId(
					cpSpecificationOption, productSpecification),
				LanguageUtils.getLocalizedMap(productSpecification.getValue()),
				GetterUtil.get(
					productSpecification.getPriority(),
					cpDefinitionSpecificationOptionValue.getPriority()),
				serviceContext);`

which will erase the value if you don't pass it which is not we expect from a PATCH

Reply:
No, if you don't pass the value map to the request there is in CPDefinitionSpecificationOptionValueModelImpl this code

public void setValueMap(
		Map<Locale, String> valueMap, Locale defaultLocale) {

		if (valueMap == null) {
			return;
		}

...
	}

that prevents the "erasing" of the value